### ISSUE TITLE
a2i_IPADDRESS rejects IPv6 scoped address (RFC 4007)

### ISSUE THREAD

**Issue by delroth** [openssl/openssl#18530]
`a2i_IPADDRESS` considers an IPv6 address with zone ID (example: `1::2%32`, `a:b:c::d%iface`) as invalid. Instead, it should likely ignore the zone ID, which is irrelevant to the "serialized" octet form of the IPv6 address. This would be the same behavior as `inet_pton` in glibc, musl, and likely other libc implementations.

See https://datatracker.ietf.org/doc/html/rfc4007 for more test cases.

**Comment by t8m** [comment]
Please note we are looking for community help implementing this issue. Please comment if you are interested in implementing this feature or if you have reasons why this feature is important enough so we should consider implementing it with higher priority than other requested features. The issue will be closed after 2 months otherwise.

**Comment by delroth** [comment]
I disagree with your classification of this bug as a feature request. If you think properly parsing IP addresses that are valid (as per RFCs) is a feature, then feel free to close this, I'm not going to be able to convince you that this is higher priority than whatever other features might be on your plate.

**Comment by t8m** [comment]
This issue was closed because it was marked with the help wanted label and there was no interest in implementing this issue or any other activity for more than 2 months.
